addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/PathTraversalScanRule.java

```diff
@@ -674,15 +689,34 @@ private static class DirNamesContentsMatcher implements ContentsMatcher {
 
         @Override
         public String match(String contents) {
-            if (contents.contains("etc") && contents.contains("bin") && contents.contains("boot")) {
-                Pattern nixDoubleCheckPattern = Pattern.compile("\\betc\\b");
-                Matcher nixDoubleCheckMatcher = nixDoubleCheckPattern.matcher(contents);
+            String result = matchNixDirectories(contents);
+            if (result != null) {
+                return result;
+            }
+            return matchWinDirectories(contents);
+        }
 
-                if (nixDoubleCheckMatcher.find()) {
-                    return "etc";
-                }
+        private String matchNixDirectories(String contents) {
+            Pattern procPattern = Pattern.compile("(^|\\W)proc(\\W|$)", Pattern.CASE_INSENSITIVE);
```

Comment:
The groups could /should be non-capturing `?:`

Reply:
Thanks! Done.